snapcraft.yaml

```diff
@@ -0,0 +1,49 @@
+name: wksctl
+version: git
+summary: Open Source Weaveworks Kubernetes Subscription
+description: >
+  wksctl allows simple creation of a Kubernetes cluster given a set of IP addresses and an SSH key. It
+  can be run in a standalone environment but is best used via a GitOps approach in which cluster and
+  machine descriptions are stored in Git and the state of the cluster tracks changes to the descriptions.
+   
+  Its features include:
+  
+  * simple creation of Kubernetes clusters
+  * manage cluster and machine descriptions using Git
+  * manage addons like Weave Net or Flux
+  * Sealed Secret integration
+confinement: devmode
+base: core18
+
+parts:
+  wksctl:
+    source: .
+    # Don't use go plugin because it doesn't seem to work with Go modules
+    plugin: nil
+    override-build: |
+      export GOBIN=$SNAPCRAFT_PART_INSTALL/bin
+      go build -o $GOBIN/wksctl ./cmd/wksctl
+    build-environment:
+      - GO111MODULE: 'on'
+      - CGO_ENABLED: '0'
+    build-packages:
+      - gcc
+      - git
+    build-snaps:
+      - go/1.12/stable
+
+plugs:
+  kube-config:
+    interface: personal-files
+    read:
+    - $HOME/.kube
+    - $HOME/.minikube
+    - $HOME/.config/k3d  # hard-coded $XDG_CONFIG_HOME for the common case
```

Comment:
Do all of these files actually need to be read? This will need an ACK from the Ubuntu Store Security people.

Aren't we reading from ~/.wks?

Reply:
Actually... kubectl will need to read the kubeconfig file there, right? wksctl will most likely just be able to write it.

Reply:
> Do all of these files actually need to be read?

I imagine the answer to that is no :) I just copied it with broad permissions to manage to compile and run and then was gonna narrow it down as part of _Verify and tighten all the interface permissions_ above. But thanks for noticing!